esp32_ulp/preprocess.py

```diff
@@ -0,0 +1,53 @@
+from . import nocomment
```

Comment:
you added a new call to `nocomment`, but I did not see you removing the original call that is already in use somewhere.

Reply:
I wanted the pre-processor to be optional - i.e. one can still run the assembler without preprocessing. That's why both the preprocessor and the assembler run nocomment.

Perhaps I can add a `remove_comments = True` arg to the Assembler.assemble() method, i.e. by default comment are removed, but when using the preprocessor before the assembler, one could specify `False` to that arg to tell the assembler to no repeat the comment removal? The input text to that method would still need to be split into lines, so I am not sure how much memory advantage we get, and so far I considered the time dimension (how long the algorithm runs) not that important.

Any suggestion how to proceed?

Reply:
not sure about how important time is, depends on how slow we are getting.
it usually is a 1 time op at program startup, so not important to be super fast, but also should not be super slow of course.

but working vs. not working (out of memory) is way more important, so be careful with memory usage. this needs practically trying on esp32, won't find this on dev pc.

Reply:
the new arg sounds like a good idea, True is correct default.

Reply:
Naming things .... :) 
```diff
 from . import opcodes
-from .nocomment import remove_comments
+from .nocomment import remove_comments as do_remove_comments
 from .util import garbage_collect

 TEXT, DATA, BSS = 'text', 'data', 'bss'
@@ -271,8 +271,8 @@ class Assembler:
                 raise ValueError('Unknown opcode or directive: %s' % opcode)
         self.finalize_sections()

-    def assemble(self, text):
-        lines = remove_comments(text)
+    def assemble(self, text, remove_comments=True):
+        lines = do_remove_comments(text) if remove_comments else text.splitlines()
         self.init(1)  # pass 1 is only to get the symbol table right
         self.assembler_pass(lines)
         self.symbols.set_bases(self.compute_bases())
```
Any suggestion for what else to call the new arg (given that `remove_comments` already is the method we want to call) or what to use instead of the alias? Or happy with the above suggestion?